Sources/Private/Experimental/Animations/ShapeAnimation.swift

```diff
@@ -77,9 +77,24 @@ extension CAShapeLayer {
   /// Adds animations for `strokeStart` and `strokeEnd` from the given `Trim` object
   @nonobjc
   func addAnimations(for trim: Trim, context: LayerAnimationContext) {
+    // `CAShapeLayer` requires that `strokeStart` be less than `strokeEnd`.
+    //  - Since this isn't a requirement in the Lottie schema, there are
+    //    some animations that have `strokeStart` and `strokeEnd` flipped.
+    //  - If we detect that this is the case for this specific `Trim`, then
+    //    we swap the start/end keyframes to match what `CAShapeLayer` expects.
+    let strokeStartKeyframes: KeyframeGroup<Vector1D>
+    let strokeEndKeyframes: KeyframeGroup<Vector1D>
+    if trim.startValueIsAlwaysGreaterThanEndValue() {
+      strokeStartKeyframes = trim.end
+      strokeEndKeyframes = trim.start
+    } else {
+      strokeStartKeyframes = trim.start
+      strokeEndKeyframes = trim.end
+    }
```

Comment:
nit: might be simpler to reason about if this method returns a tuple of the start/end keyframes:
```suggestion
    let (strokeStartKeyframes, strokeEndKeyframes) = trim.caShapeLayerKeyframes()
```

Reply:
ooohh good idea